content/articles/understanding-when-to-use -jsp-and-when-to-use-servlet/index.md

```diff
@@ -0,0 +1,184 @@
+2. Tomcat or any equivalent web server provides the `_jspService`, `doPost` or `doGet` method, as well as the `request` and `response` parameters as shown:
+   ```Java
+   public void _jspService(HttpServletRequest request, HttpServletResponse response)                        
+   {
+      PrintWriter out = response.getWriter();
+      response.setContentType("text/html");
+      out.write("<html><body>");
+      out.write("Hello world");
+      out.write("</body></html>");
+   }
+   ```
+3. The JSP code must be enclosed in one of the tags listed below
+- Scriptlet tag(<% code %>) 
+- declaration tag(<%! declaration %>)
+- Directive Tag(<%@ page ... %>)
+- Expression tag(<%= output%> )
+
+We will presume that you are familiar with JSP tags and will not describe them in this article. The JSP code contained within the following JSP tags will be copied into the Servlet page, as follows: 
+- The content of the scriptlet tag will be copied into the corresponding Servlet `_jspService()` method
+- The content of the declaration tag will be copied outside of the associated Servlet page's `_jspService()` method, and
+- All Java imports are contained within directive tags in JSP, and the contents of these tags are copied outside of the resulting `Servlet` class
+  
+### Example
+To explain how JSP gets converted to the matching `Servlet` during code execution, we'll use some simple JSP code.
+
+#### Steps to take
+
+#### Step 1: Create a new project
+Using Netbeans IDE click on File -> New Project -> Java Web -> Web Application and select next. Last but not least, give the project a name and click finish. In my case, I'll call it **JSPToServlet**. Finally, choose an appropriate server and click finish. In my case, I'll be using Tomcat version 9. 
+
+#### Step 1: Create JSP file
+
+To create a new JSP file, right-click on the project name, choose JSP, and then name it `new_jsp` as shown.
+
+```JSP
+    </head>
+    <body>
+      <%
+       out.print("Hello World");
+      %>
+    </body>
+```
+#### Generated Servlet code
+
+Run the project and right-click on the JSP file and select view Servlet:
+
+```Java
+import javax.servlet.*;
+import javax.servlet.http.*;
+import javax.servlet.jsp.*;
+
+public final class newjsp_jsp extends org.apache.jasper.runtime.HttpJspBase
+    implements org.apache.jasper.runtime.JspSourceDependent,
+                 org.apache.jasper.runtime.JspSourceImports {
+
+  public void _jspService(final javax.servlet.http.HttpServletRequest request, final javax.servlet.http.HttpServletResponse response) 
+	  {
+      javax.servlet.jsp.JspWriter out = response.getJspWriter();
+      out.write("    <body>\n");
+      out.write("      ");
+
+       out.print("Hello World");
+      
+      out.write("\n");
+      out.write("    </body>\n");
+      out.write("</html>\n");
+  }
+}
+

```

Comment:
I seem to get a different Servlet output as seen in the file below (I put it into Markdown just because you can't add Java files to Github comments):

[output.md](https://github.com/section-engineering-education/engineering-education/files/8575739/output.md)
 
I guess that your IDE doesn't show the exact compiled JSP but rather a simplified version of it. I was using IntelliJ as my IDE and got that compiled JSP from where my local tomcat server puts it. You should address this in some way and update your article appropriately.